test/idp_metadata_parser_test.rb

```diff
@@ -50,9 +50,37 @@ class MockResponse
     it "accept self signed certificate if insturcted" do
       idp_metadata_parser = OneLogin::RubySaml::IdpMetadataParser.new
       settings = idp_metadata_parser.parse_remote(@url, false)
-
       assert_equal OpenSSL::SSL::VERIFY_NONE, @http.verify_mode
     end
   end
 
+  describe "download and parse IdP descriptor file (HTTP)" do
+    before do
+      mock_response = MockResponse.new
+      mock_response.body = idp_metadata_2
```

Comment:
Please consider more descriptive names _(or at the very least, use comments)_ for things like `idp_metadata_2`.